dependency/test/entrypoint

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env bash
+
+set -eu
+set -o pipefail
+
+readonly TESTDIR="/test"
+readonly PARENTDIR="$(cd "$(dirname "$0")" && pwd)"
+
+function main() {
+  local tarball expected_version
+
+  while [ "${#}" != 0 ]; do
+    case "${1}" in
+      --tarballPath)
+        tarball="${2}"
+        shift 2
+        ;;
+
+      --version)
+        expected_version="${2}"
+        shift 2
+        ;;
+
+      "")
+        shift
+        ;;
+
+      *)
+        echo "unknown argument \"${1}\""
+        exit 1
+    esac
+  done
+
+  if [[ -z "${tarball:-}" ]]; then
+    echo "--tarballPath is required"
+    exit 1
+  fi
+
+  if [[ -z "${expected_version:-}" ]]; then
+    echo "--version is required"
+    exit 1
+  fi
+
+  echo "Unpacking tarball..."
+  tar zxf "${tarball}"
+
+  echo "Checking tarball structure..."
+
+  if ! [ -f "${TESTDIR}/bin/ruby" ];
+  then
+    echo "missing ruby executable"
+    ls -al "${TESTDIR}"
+    exit 1
+  fi
+
+  echo "Checking ruby version..."
+
+  actual_version=$(${TESTDIR}/bin/ruby --version | cut -d' ' -f2 | sed 's/p.*//')
+  if [[ "${actual_version}" != "${expected_version}" ]]; then
+    echo "version ${actual_version} does not match expected version ${expected_version}"
+    exit 1
+  fi
+
+  # test installing a gem
```

Comment:
Does this comment reflect the steps taken by the test? It looks like the test app doesn't install a gem, but I might be missing something.

Reply:
bad copy paste from dep-server code I was using :)